packages/binding-parser/src/ast/binding-parser-visitor.ts

```diff
@@ -0,0 +1,201 @@
+import { CstNode, IToken } from "chevrotain";
+import type { Position } from "vscode-languageserver-types";
+import {
+  ARRAY,
+  BOOLEAN_VALUE,
+  COLON,
+  COMMA,
+  KEY,
+  LEFT_SQUARE,
+  LEFT_CURLY,
+  NULL_VALUE,
+  NUMBER_VALUE,
+  OBJECT,
+  OBJECT_ITEM,
+  RIGHT_SQUARE,
+  RIGHT_CURLY,
+  STRING_VALUE,
+  VALUE,
+  TEMPLATE,
+} from "../constant";
+import { createToken } from "../utils/create";
+import { locationToRange } from "../utils/range";
+import type {
+  StructureElement,
+  CollectionValue,
+  PrimitiveValue,
+  StructureValue,
+  Value,
+  VisitorParam,
+  CreateToken,
+  ObjectChildren,
+  Template,
+  BindingNode,
+  TemplateChildren,
+  TokenType,
+  ObjectItemChildren,
+  ValueChildren,
+  ArrayChildren,
+} from "../types/binding-parser";
+import { bindingParser } from "../parser/binding-parser";
+
+const BaseVisitor = bindingParser.getBaseCstVisitorConstructor();
+class BindingParserVisitor extends BaseVisitor {
+  startPosition?: Position;
+  constructor(startPosition?: Position) {
+    super();
+    this.startPosition = startPosition;
+    this.validateVisitor();
+  }
+  visit(cstNode: CstNode): BindingNode {
+    return super.visit(cstNode, { location: cstNode.location } as VisitorParam);
+  }
+  [TEMPLATE](node: TemplateChildren): Template {
+    const objects = node[OBJECT] ?? [];
+    const spaces = [];
+    const ast: Template = {
+      bindings: [],
+      spaces,
+      type: "template",
+    };
+    for (const data of objects) {
+      const result = this.visit(data) as StructureValue;
+      ast.bindings.push(result);
+    }
+    return ast;
+  }
+  [OBJECT](node: ObjectChildren, param: VisitorParam): StructureValue {
+    const leftCurly = this.createToken(LEFT_CURLY, param, node[LEFT_CURLY]);
+    const rightCurly = this.createToken(RIGHT_CURLY, param, node[RIGHT_CURLY]);
+    const elements: StructureElement[] = [];
+    const data = node[OBJECT_ITEM] ?? [];
+    for (const element of data) {
+      const result = this.visit(element) as StructureElement;
+      if (result.key || result.colon || result.value) {
+        elements.push(result);
+      }
+    }
+    const range = locationToRange({ ...param, position: this.startPosition });
+    const commas = this.getCommas(param, node[COMMA]);
+    return {
+      leftCurly,
+      elements,
+      rightCurly,
+      range,
+      commas,
+      type: "structure-value",
+    };
+  }
+  [OBJECT_ITEM](
+    node: ObjectItemChildren,
+    param: VisitorParam
+  ): StructureElement {
+    const key = this.createToken(KEY, param, node[KEY] ?? node[STRING_VALUE]);
```

Comment:
Can't we handle the quotes already here? It's rather error prone to have to check this every time when you try to use the key value.